### ISSUE TITLE
Dependency Resolvings and usings showing as unnecessary after update

### ISSUE THREAD

**Issue by CryoViking** [seblj/roslyn.nvim#340]
### Description

After upgrading to the latest roslyn LSP through mason and the latest nvim-roslyn plugin version, namespaces are being marked as "Unnecessary" when it's not true, and the issue persists in razor files with the rzls co-hosted LSP.

I've also noticed that components from these ``@usings`` in razor files also show as "Found markup element with unexpected name..."

<img width="809" height="363" alt="Image" src="https://github.com/user-attachments/assets/ce70b889-2699-430f-b48f-ea5d0ba4c327" />

<img width="2560" height="628" alt="Image" src="https://github.com/user-attachments/assets/12b34923-3a0c-4422-a7ee-6b01472929f2" />

I've also noticed that sometimes whether I open a .cs or a .razor file first can sometimes have a random effect on whether or not a file of the other type opened second works or not. Or at least doesn't show the LSP warnings/errors.

And for reference, this is the file structure:

<img width="490" height="1237" alt="Image" src="https://github.com/user-attachments/assets/98f3c570-3d00-4a0e-bbc2-0e8723c206a4" />

### Neovim version

NVIM v0.12.1

### Operating system and version

macOS 26.3.1

### Expected behavior

I would expect the behaviour to show the namespaces as resolved/found/necessary and for the razor components to be known/found.

### Actual behavior

As mentioned in the bug description:

<img width="1157" height="363" alt="Image" src="https://github.com/user-attachments/assets/f1d79b56-f17e-4a9d-aa8a-55db19318fff" />

<img width="2560" height="469" alt="Image" src="https://github.com/user-attachments/assets/b15d7877-e294-487c-9f1d-46cf4df06fc3" />

### Minimal config

```Lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

require("lazy.minit").repro({
    spec = {
        {

            "mason-org/mason.nvim",
            opts = {
                registries = {
                    "github:mason-org/mason-registry",
                    "github:crashdummyy/mason-registry",
                },
            },
        },
        {
            "seblyng/roslyn.nvim",
            init = function()
                vim.filetype.add({
                    extension = {
                        razor = "razor",
                        cshtml = "razor",
                    },
                })
            end,
        },
    },
})
```

### Roslyn checkhealth

```text
==============================================================================
roslyn:                                                                     ✅

roslyn.nvim: Requirements ~
- ✅ OK Neovim >= 0.11
- ✅ OK dotnet SDK >= 10 (found 10.0.100)

roslyn.nvim: Roslyn Language Server ~
- ✅ OK /Users/cryosis/.local/share/nvim/mason/bin/roslyn: found
- ✅ OK Razor extension: found at /Users/cryosis/.local/share/nvim/mason/packages/roslyn/libexec/.razorExtension
- ✅ OK vscode-html-language-server: found
- ✅ OK html-lsp client: configured

roslyn.nvim: File Watching Configuration ~
- File watching: auto (using Neovim's file watcher)

roslyn.nvim: Solution Detection ~
- ✅ OK Selected solution: /Users/cryosis/HouseHlaalu/EnchantisAutomata/ShadowMirror/ShadowMirror.sln
```

**Comment by tris203** [comment]
There is specially as 0.12 branch

https://github.com/seblyng/roslyn.nvim/pull/330

Could you test with that please

**Comment by CryoViking** [comment]
What's the minimum path for testing with a branch from a fork of a plugin in my config?

**Comment by CryoViking** [comment]
For now I've just got:
```lua
  {
    "tris203/roslyn.nvim",
    branch = "0.12",
    ---@module 'roslyn.config'
    ---@type RoslynNvimConfig
    ...
  }
```

I think it seems to work at first glance, but will use extensively today to see if it does work.
